src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -1723,6 +1723,12 @@ private void AddAdditionalTraceContext(MapField<string, string> attributes, Scri
             {
                 Activity.Current?.AddTag(ResourceSemanticConventions.FaaSName, context.FunctionMetadata.Name);
                 Activity.Current?.AddTag(ResourceSemanticConventions.FaaSTrigger, OpenTelemetryConstants.ResolveTriggerType(context.FunctionMetadata?.Trigger?.Type));
+                if (context.FunctionMetadata?.Trigger?.Raw != null && context.FunctionMetadata.Trigger.Type == "httpTrigger" && context.FunctionMetadata.Trigger.Raw.TryGetValue("route", StringComparison.OrdinalIgnoreCase, out var value))
+                {
+                    string routeValue = value.ToString();
+                    Activity.Current.DisplayName = $"{Activity.Current.DisplayName} {routeValue}";
```

Comment:
is `Activity.Current` guaranteed here?
Also, wonder if we can rely on it's value being http method (something sets it to `{METHOD}`, but what if it starts to set route too? Or, on older .NET versions it'd likely default to activity name, which would be `HttpRequestIn` or something not so friendly.

The tl;dr: do we have method name around? can we format the display name that's guaranteed to comply?

Reply:
Thanks for taking a look @lmolkova :)

The Host that executes this part of the code for httpTriggers is a .NET 8 Web host, so I'm pretty sure if the activity exists and the trigger is an httpTrigger we can count on it being correct.

I hope @RohitRanjanMS can correct me if I'm wrong.

I'll add a null check on the Activity.Current, I missed that one :)

Reply:
Thanks, @stebet , I will have a look at this today.

Reply:
right, so what if in the future version Functions would use routing on the worker? Also, if I'm not mistaken, on .NET 8 OTel instrumentations sets ASP.NET Core activity name, but in .NET 9 (or 10, don't remember), it will be the ASP.NET Core itself. Plus some users may enable ASP.NET Core instrumentation and others won't, so it'd be safer to format the whole thing and and not rely on DisplayName matching method.

Reply:
End users don't have control of this host process except through host.json config AFAIK. It'll always be at the root since it handles the reverse proxying and routing of incoming function requests.

Reply:
We can add this logic in `EnrichWithHttpResponse`. You can use IRoutingFeature to get the route information.

```.AddAspNetCoreInstrumentation(o =>
{
    o.EnrichWithHttpResponse = (activity, httpResponse) =>
    {
        var routingFeature = httpResponse.HttpContext.Features.Get<AspNetCore.Routing.IRoutingFeature>();
        // Set the route and display name here
    };
});